### PR TITLE
Feat: API Extensions uncommon queries

### PR DESCRIPTION
**Pull request by eduardoformiga**
## What's the purpose of this pull request?

This PR aims to enable the possibility of using API Extension for some queries that weren't mapped in the first 'wave'.

Queries: `ClientShippingSimulationQuery`, `ClientSearchSuggestionsQuery`, `ClientTopSearchSuggestionsQuery`. 
It also facilitates the logic in the `ShippingSimulation` component and exposes the address as prop.

### Related Issue
- https://github.com/vtex/faststore/issues/2046

## How it works?

It integrates the default fragments in the corresponding queries and exposes the data from the hooks and function that fetches these data.

## How to test it?

1. add new fields into your schema using [API Extensions](https://faststore-site-git-feat-api-extensions-other-queries-faststore.vercel.app/docs/api-extensions) or just add existing fields returned from Faststore API into the [mapped fragments](https://faststore-site-git-feat-api-extensions-other-queries-faststore.vercel.app/docs/api-extensions#step-2-define-the-graphql-fragment). 
2. Use the experimental functions or hooks while [overriding components](https://faststore-site-git-feat-api-extensions-other-queries-faststore.vercel.app/docs/building-sections/overriding-components-and-props#step-2---setting-up-the-override).

e.g.

- `ClientShippingSimulation` (fragment) exposes fields into `ClientShippingSimulationQuery`. Then import and use the `useShippingSimulation_unstable()`, or the `getShippingSimulation_unstable()` function from `@faststore/core/experimental` in custom components.

- `ClientSearchSuggestions` (fragment) exposes fields into `ClientSearchSuggestionsQuery`. Then import and use the `useSuggestions_unstable()` from `@faststore/core/experimental` in custom components.

- `ClientTopSearchSuggestions` (fragment) exposes fields into `ClientTopSearchSuggestionsQuery`. Then import and use the `useTopSearch_unstable()` from `@faststore/core/experimental` in custom components.

### CODE REVIEW

File: packages/core/src/customizations/fragments/ClientSearchSuggestions.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import { gql } from '@faststore/graphql-utils'
+
+export const fragment = gql`
+  fragment ClientSearchSuggestions on Query {
+    search(first: 5, term: $term, selectedFacets: $selectedFacets) {
```
Comment (icazevedo) [inline packages/core/src/customizations/fragments/ClientSearchSuggestions.ts:5]: Allowing to customize the amount of items will surely be a task in the future. Not sure if we should do something about it now, tho.

File: packages/core/src/sdk/search/useSuggestions.ts [hunk 2]
```diff
@@ -74,6 +75,7 @@ function useSuggestions(term: string, limit: number = MAX_SUGGESTIONS) {
   })
 
   return {
+    data,
     terms: (data?.search.suggestions.terms ?? []).slice(0, limit),
     products: (data?.search.suggestions.products ?? []).slice(0, limit),
```
Comment (icazevedo) [inline packages/core/src/sdk/search/useSuggestions.ts:80]: does still makes sense to return these when we're already returning data?

File: packages/core/src/sdk/search/useTopSearch.ts [hunk 3]
```diff
@@ -41,6 +42,7 @@ function useTopSearch(
   })
 
   return {
+    data,
     terms: (data?.search.suggestions.terms ?? initialTerms).slice(0, limit),
```
Comment (icazevedo) [inline packages/core/src/sdk/search/useTopSearch.ts:46]: Same here

**Comment by icazevedo** [comment]
Also, mention the issue here so github links it there.

File: packages/core/src/customizations/fragments/ClientSearchSuggestions.ts (hunk 1 above)
Comment (eduardoformiga) [reply packages/core/src/customizations/fragments/ClientSearchSuggestions.ts:5]: this is just the default behavior. So when people use the starter they will override this file. But we can use a variable instead, actually.

File: packages/core/src/sdk/search/useSuggestions.ts (hunk 2 above)
Comment (eduardoformiga) [reply packages/core/src/sdk/search/useSuggestions.ts]: I was thinking about that and [change it here](https://github.com/vtex/faststore/pull/2051/commits/db48b37e20f78a0e5fd21e9165b5bade6720d9c2).

File: packages/core/src/sdk/search/useTopSearch.ts (hunk 3 above)
Comment (eduardoformiga) [reply packages/core/src/sdk/search/useTopSearch.ts]: the same [here](https://github.com/vtex/faststore/pull/2051/commits/97fb5fdc95aabdfbd5a11f51a2aa090435ffa5fe).

**Review by emersonlaurentino — APPROVED** [review]

**Pull request merged**